internal/login/command.go

```diff
@@ -300,16 +284,44 @@ func (c *command) loginMDS(cmd *cobra.Command, url string) error {
 	return nil
 }
 
-func getCACertPath(cmd *cobra.Command) (string, error) {
-	if caCertPath, err := cmd.Flags().GetString("ca-cert-path"); caCertPath != "" || err != nil {
-		return caCertPath, err
+// Current functionality:
+// empty ca-cert-path is equivalent to not using ca-cert-path flag
+// if users want to login with ca-cert-path they must explicitly use the flag every time they login
+//
+// For legacy users:
+// if ca-cert-path flag is not used, then return caCertPath value stored in config for the login context
+// if user passes empty string for ca-cert-path flag then reset the ca-cert-path value in config for the context
+// (only for legacy contexts is it still possible for the context name without ca-cert-path to have ca-cert-path)
+func (c *command) getCACertPath(cmd *cobra.Command, username, url string) (string, bool, error) {
+	caCertPath, err := cmd.Flags().GetString("ca-cert-path")
+	if err != nil {
+		return "", false, err
+	}
+
+	if caCertPath == "" {
+		caCertPath = pauth.GetEnvWithFallback(pauth.ConfluentPlatformCACertPath, pauth.DeprecatedConfluentPlatformCACertPath)
+	}
+
+	var isLegacyContext bool
+	if caCertPath == "" {
+		contextName := pauth.GenerateContextName(username, url, "")
+		caCertPath = c.checkLegacyContextCACertPath(cmd, contextName)
```

Comment:
This also seems like something we could remove in v4

Reply:
Yup, there's v4 tickets for this and the deprecated environment variables.